src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -16,7 +16,8 @@
     <!-- .NETCoreApp == 5.0 -->
     <_MaxSupportedLangVersion Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' == '5.0' AND
                                          '$(_MaxSupportedLangVersion)' == ''">9.0</_MaxSupportedLangVersion>
-    
+
+    <MaxSupportedLangVersion>$(_MaxSupportedLangVersion)</MaxSupportedLangVersion>
```

Comment:
Why not just directly assign to the public property? What benefit does the _ version get you?

Reply:
Other things are now taking a dependency on the _ version. We can't break them either. This way we can figure out the logic 'privately' and expose the result 'publically'

Reply:
Ah. :smile: This may deserve a comment in the code.